### PR TITLE
Added UA20160912: Revised Ranger.

### PR DESCRIPTION
**Pull request by NickVendel**
__**Revised Ranger**__ was adapted into a class.
**Few details and notes:**
• I removed `<Ranger Conclave>` supports of Ranger Archetypes from XGTE, because there's no feature for level 5 in them (which is _Coordinated Attack_ in Beast Conclave and _Extra Attack_ in other ones), making this support useless.
• I added **Oozes** and **Plants** as basic _Favored Enemies_, since they're nowhere to be found in original UA pdf.
• **Animal Companion feature is missing rules.** Right now there's no way to implement these rules. I will update file as soon as these rules are included into Aurora Builder. So, for the time being, **app applies Proficiency Bonus only to AC**, everything else, like Attack Bonus, Damage, Skills, Saving Throws, etc., must be calculated manually.

### CODE REVIEW

File: unearthed-arcana/2016/20160912.xml [hunk 1]
```diff
@@ -0,0 +1,914 @@
...
+            <div element="ID_WOTC_UA20160912_CLASSFEATURE_RANGER_PRIMEVAL_AWARENESS" />
+            <div element="ID_WOTC_UA20160912_CLASSFEATURE_RANGER_ASI" />
+            <div element="ID_WOTC_UA20160912_CLASSFEATURE_RANGER_GREATER_FAVORED_ENEMY" />
+            <div element="ID_WOTC_UA20160912_CLASSFEATURE_RANGER_FLEET_OF_FOOT" />
+            <div element="ID_WOTC_UA20160912_CLASSFEATURE_RANGER_HIDE_IN_PLAIN_SIGHT" />
+            <div element="ID_WOTC_UA20160912_CLASSFEATURE_RANGER_VANISH" />
+            <div element="ID_WOTC_UA20160912_CLASSFEATURE_RANGER_FERAL_SENSES" />
+            <div element="ID_WOTC_UA20160912_CLASSFEATURE_RANGER_FOE_SLAYER" />
+        </description>
+        <sheet alt="Shelter of the Faithful" />
```
Comment (swdriessen) [inline unearthed-arcana/2016/20160912.xml]: `<sheet alt="Ranger" />`

File: unearthed-arcana/2016/20160912.xml [hunk 2]
```diff
@@ -0,0 +1,914 @@
...
+            <grant type="Class Feature" name="ID_WOTC_UA20160912_CLASSFEATURE_RANGER_PRIMEVAL_AWARENESS" level="3"/>
+            <grant type="Class Feature" name="ID_WOTC_UA20160912_CLASSFEATURE_RANGER_ASI" level="4"/>
+            <grant type="Class Feature" name="ID_WOTC_UA20160912_CLASSFEATURE_RANGER_GREATER_FAVORED_ENEMY" level="6"/>
+            <grant type="Class Feature" name="ID_WOTC_UA20160912_CLASSFEATURE_RANGER_FLEET_OF_FOOT" level="8"/>
+            <grant type="Class Feature" name="ID_WOTC_UA20160912_CLASSFEATURE_RANGER_HIDE_IN_PLAIN_SIGHT" level="10"/>
+            <grant type="Class Feature" name="ID_WOTC_UA20160912_CLASSFEATURE_RANGER_VANISH" level="14"/>
+            <grant type="Class Feature" name="ID_WOTC_UA20160912_CLASSFEATURE_RANGER_FERAL_SENSES" level="18"/>
+            <grant type="Class Feature" name="ID_WOTC_UA20160912_CLASSFEATURE_RANGER_FOE_SLAYER" level="20"/>
+        </rules>
+        <multiclass id="ID_WOTC_PHB_MULTICLASS_RANGER">
```
Comment (swdriessen) [inline unearthed-arcana/2016/20160912.xml]: use the `ID_WOTC_UA20160912_MULTICLASS_RANGER` UA ID here instead of overriding PHB ID

**Comment by swdriessen** [comment]
If you want you can cleanup the file and comment the content out, then i'll merge it and combine it with new release pull request

**Comment by NickVendel** [comment]
Idk, maybe i should clean it up for the next one. I will comment when i update it

**Comment by omerbl1996** [comment]
Do you guys know if one day there will be a possibility to make all ranger archetypes possible for the revised ranger?

**Comment by NickVendel** [comment]
It's possible, problem is that Revised Ranger gains Extra Attack Feature from it's Archetypes, so Revised Ranger with normal Ranger Archetype would just miss Extra Attack.

**Comment by omerbl1996** [comment]
But how can I do it? I have no idea lol

**Comment by swdriessen** [comment]
It will be merged once the file is cleaned up, then you don't have to worry about the file itself. When it is included I will provide an example snippet/file to extend it to allow the core/supplement ranger archetypes to be listed when selecting a ranger conclave. (and perhaps a link in the source to said snippet for future reference)

_Note: the Gloom Stalker for example currently works out of the box because I played the revised ranger in the summer of 2017. These archetypes come from my personal files and i probably had them modified when XGTE came out and forgot to remove the entries. So you have to also do this for all archetypes then, including the ones that currently 'just work'._

**Comment by omerbl1996** [comment]
So basically what you're saying is that it'll take time but when you'll be done arranging the files you'll be able to merge all the archetypes normally?

**Comment by swdriessen** [comment]
It's not supported out of the box because they were not made to be compatible with each other.

But once it's merged, you can add a single custom file (which will be provided) that just makes all archetypes work with this revised class.

**Comment by omerbl1996** [comment]
So when it'll merge the Aurora will be updated automatically?

**Comment by swdriessen** [comment]
Apart from that 'single custom file', yes. _Merged_ means that the files are included and the builder will automatically download them. 

Since you're notified on this pull request you'll get a notification when that's done.

**Comment by erjicles** [comment]
Any update on when this will be available?

**Comment by NickVendel** [comment]
@erjicles hard to say, i wouldn't recommend waiting, either just grab file for it here or better find your way to Discord and my post in #unearthed-arcana (or on reddit, though i don't remember if i updated that version...), since that version removes supports for Ranger Conclave within the file, rather than downloading all three from here.

**Comment by erjicles** [comment]
> @erjicles hard to say, i wouldn't recommend waiting, either just grab file for it here or better find your way to Discord and my post in #unearthed-arcana (or on reddit, though i don't remember if i updated that version...), since that version removes supports for Ranger Conclave within the file, rather than downloading all three from here.

Thanks. I went ahead and copied the file from here and put it into my custom user files folder. Is the file from your Discord post any better?

**Comment by swdriessen** [comment]
I'll look into finishing it myself if i have the time next weekend or early next month.

**Comment by swdriessen** [comment]
This content repository is being archived. I am closing any pending issues and pull requests.
I'll be sharing more information about concrete changes at a later date on the website.

**Pull request closed without merging**